### PR TITLE
Code Review of Q12

### PR DESCRIPTION
**Pull request by Harshwardhanwtagline**

### CODE REVIEW

File: Q12.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        #Checking the pin on bases of condition and then giving account information
+        if self.pin == pin:
+            return "Account Holder Name: {}\nAccount Number: {}\nTotal Balance: {}".format(self.acc_name, self.acc_number, self.balance)
+        return "Please Enter Correct Pin"
+    
+    def deposit(self, pin, amount):
+
+        #Checking the pin on bases of condition and then depositing money in  account.
+        if (self.pin == pin):
+            self.balance += amount
```
Comment (priteshtagline) [inline Q12.py:18]: Why you are repeating this lines multiple times? You again didn't checked your code properly. You've already created check_pin function but same logic you are writing here. Please update it.

File: Q12.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            return "Your account don't have sufficient balance"
+        return "Please Enter Correct Pin"
+    
+    def __str__(self) -> str:
+        return "Account Holder Name: {}\nAccount Number: {}\nTotal Balance: {}".format(self.acc_name, self.acc_number, self.balance)
+
+        
+
+
+
```
Comment (priteshtagline) [inline Q12.py:38]: Remove these extra spaces.

File: Q12.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+class BankAccount:
+    def __init__(self, acc_name, acc_number, balance, pin):
+        self.acc_name = acc_name
+        self.acc_number=acc_number
+        self.balance=balance
```
Comment (priteshtagline) [inline Q12.py:5]: The space before and after `=` should be consistent. For above lines, you have specified space but here you haven't.

File: Q12.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        self.pin=pin
+    
+    def check_pin(self, pin):
+        #Checking the pin on bases of condition and then giving account information
+        if self.pin == pin:
+            return "Account Holder Name: {}\nAccount Number: {}\nTotal Balance: {}".format(self.acc_name, self.acc_number, self.balance)
```
Comment (himanshu-suthar-tagline) [inline Q12.py]: Why you are using same code to check pin again and again if you have created check_pin function? Please update it.

File: Q12.py (hunk 2 above)
Comment (himanshu-suthar-tagline) [inline Q12.py:36]: Remove unnecessary spaces.

File: Q12.py (hunk 3 above)
Comment (himanshu-suthar-tagline) [inline Q12.py]: There should be one space before and after `=`, for the first, you have speicified but for others you didn't specified a space.

**Pull request merged**